### PR TITLE
[C-2914] USDC purchase options for new upload UI (web)

### PR DESCRIPTION
**Pull request by amendelsohn**
### Description
Screenshot 2023-08-15 at 1 52 01 PM

TODOs:
- Move descriptions of radio options to checkedContent
- Input validation
- Cap preview at length of track -15s
- Cap price at some max value

### Dragons

Need to make sure InputV2 changes don't break other usages

### Feature Flags ###

`UPLOAD_REDESIGN_ENABLED` & `USDC_PURCHASES`

### CODE REVIEW

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/amendel-usdc-upload-options-web

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/amendel-usdc-upload-options-web

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/amendel-usdc-upload-options-web

**Review by dylanjeffers — APPROVED** [review]
wonderful job! all looks real clean and makes good sense. are we going to add the validation for the value range for preview to the zod schema? i was able to go through with 0.00 for sale amount and some bad data for preview time. also def curious your thoughts about the strategy around component composition

File: packages/web/src/pages/upload-page/fields/availability/UsdcPurchaseFields.tsx [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+type TrackAvailabilityFieldsProps = {
+  disabled?: boolean
+}
+
+export const UsdcPurchaseFields = (props: TrackAvailabilityFieldsProps) => {
+  const { disabled } = props
+  const [, , { setValue: setPrice }] = useField(PRICE)
+
+  const handleChange: ChangeEventHandler<HTMLInputElement> = useCallback(
```
[the remark this answers is not in the document]
Comment (amendelsohn) [reply packages/web/src/pages/upload-page/fields/availability/UsdcPurchaseFields.tsx:47]: yeah I'll circle back to the rest of validation I think

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/amendel-usdc-upload-options-web

**Pull request merged**